Sources/Core/ObjectiveCDictionaryExtension.swift

```diff
@@ -0,0 +1,152 @@
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[\(dictionary) setObject: [[NSValueTransformer valueTransformerForName:\(dateValueTransformerKey)] transformedValue:\(propIVarName)]  forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[\(dictionary) setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .enumT(.integer):
+            return "[\(dictionary) setObject: [NSNumber numberWithInteger:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .enumT(.string):
+            return "[\(dictionary) setObject: "+enumToStringMethodName(propertyName: param, className: self.className) + "(\(propIVarName))" + " forKey: @\"" + param + "\" ];"
+        case .array(itemType: let itemType?):
+            let currentResult = "result\(counter)"
+            let currentObj = "obj\(counter)"
+            switch itemType {
+            case .reference,
+                .object:
+                return ObjCIR.scope {[
+                    "NSArray *items = \(propIVarName);",
+                    "NSMutableArray *\(currentResult) = [NSMutableArray arrayWithCapacity:items.count];",
+                    ObjCIR.forStmt("id \(currentObj) in items") { [
+                        ObjCIR.ifStmt("\(currentObj) != (id)kCFNull") { [
+                             "[\(currentResult) addObject:[ \(currentObj) dictionaryRepresentation] ];"
+                            ]}
+                        ]},
+                        "[\(dictionary) setObject:\(currentResult) forKey: @\"" + param + "\" ];"
+                    ]}
+            default:
+            return ObjCIR.scope {[
+                "NSArray *items = \(propIVarName);",
+                "NSMutableArray *\(currentResult) = [NSMutableArray arrayWithCapacity:items.count];",
+                ObjCIR.forStmt("id \(currentObj) in items") { [
+                    ObjCIR.ifStmt("\(currentObj) != (id)kCFNull") { [
+                        "[\(currentResult) addObject: \(currentObj)];"
+                        ]}
+                    ]},
+                "[\(dictionary) setObject: "+currentResult + " forKey: @\"" + param + "\" ];"
+                ]}
+            }
+        case .map(valueType: let type):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                            renderAddObjectStatement(param, type!, dictionary)
+                            ]},
+                ObjCIR.elseStmt({[
+                    "[\(dictionary) setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .oneOf(types: let avTypes):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+
+            ObjCIR.switchStmt("\(propIVarName).internalType") {
```

Comment:
This implementation should simply call the `dictionaryRepresentation` of the ADT instance since you've added an implementation you can leverage here.